### PR TITLE
feat(graphs): add GridsMask node attribute builder

### PR DESCRIPTION
**Pull request by JPXKQX**
## Description

Add a new node attribute builder, `GridsMask`.

## What problem does this change solve?

This PR enables more customisation for regional models.

## What issue or task does this change relate to?

#350 

##  Additional notes ##

One limitation of the current solution is that this node attribute builder only supports cases with 2 datasets.

***As a contributor to the Anemoi framework, please ensure that your changes include unit tests, updates to any affected dependencies and documentation, and have been tested in a parallel setting  (i.e., with multiple GPUs). As a reviewer, you are also responsible for verifying these aspects and requesting changes if they are not adequately addressed. For guidelines about those please refer to https://anemoi.readthedocs.io/en/latest/***

----
📚 Documentation preview 📚: https://anemoi-training--355.org.readthedocs.build/en/355/

----
📚 Documentation preview 📚: https://anemoi-graphs--355.org.readthedocs.build/en/355/

----
📚 Documentation preview 📚: https://anemoi-models--355.org.readthedocs.build/en/355/

### CODE REVIEW

File: graphs/src/anemoi/graphs/nodes/attributes/masks.py [hunk 1]
```diff
@@ -50,13 +50,57 @@ def get_raw_values(self, nodes: NodeStorage, **kwargs) -> torch.Tensor:
...
+
+    def __init__(self, grids: int | list[int] = 0) -> None:
+        super().__init__()
+        self.grids = [grids] if isinstance(grids, int) else grids
+
+    def get_raw_values(self, nodes: NodeStorage, **kwargs) -> torch.Tensor:
+        grid_sizes = self.get_grid_sizes(nodes)
+        mask = torch.zeros(sum(grid_sizes), dtype=torch.bool)
+        for grid_id in self.grids:
+            start_grid = 0 if grid_id == 0 else grid_sizes[grid_id - 1]
```
Comment (sabrinawahl) [inline graphs/src/anemoi/graphs/nodes/attributes/masks.py]: To work with more than two datasets it should be:
`start_grid = 0 if grid_id == 0 else sum(grid_sizes[:grid_id])`

File: graphs/src/anemoi/graphs/nodes/attributes/masks.py [hunk 2]
```diff
@@ -50,13 +50,57 @@ def get_raw_values(self, nodes: NodeStorage, **kwargs) -> torch.Tensor:
...
+            end_grid = grid_sizes[grid_id - 1]
```
Comment (sabrinawahl) [inline graphs/src/anemoi/graphs/nodes/attributes/masks.py]: This line should be:
`end_grid = grid_sizes[0] if grid_id == 0 else sum(grid_sizes[:grid_id+1])`

File: graphs/src/anemoi/graphs/nodes/attributes/masks.py [hunk 3]
```diff
@@ -50,13 +50,57 @@ def get_raw_values(self, nodes: NodeStorage, **kwargs) -> torch.Tensor:
...
+    Methods
+    -------
+    compute(self, graph, nodes_name)
+        Compute the attribute for each node.
+    """
+
+    def __init__(self, grids: int | list[int] = 0) -> None:
+        super().__init__()
+        self.grids = [grids] if isinstance(grids, int) else grids
+
```
Comment (sabrinawahl) [inline graphs/src/anemoi/graphs/nodes/attributes/masks.py]: Maybe this will work as well and we don't have to struggle with the indices:
```
    def get_raw_values(self, nodes: NodeStorage, **kwargs) -> torch.Tensor:
        grid_sizes = self.get_grid_sizes(nodes)
        id_mask = torch.tensor([i for i, n in enumerate(grid_sizes) for _ in range(n)], dtype=torch.int32)
        bool_mask = torch.isin(id_mask, torch.tensor(grids))
        return bool_mask
```
